### PR TITLE
chore: Update serving setup to use common scripts

### PR DESCRIPTION
**Pull request by dsimansk**
It will either be working like a charm or fail horribly.

/cc @navidshaikh

### CODE REVIEW

**Comment by dsimansk** [comment]
Let's add `yq` to CI image.
```
/tmp/tmp.QN2IATbx20/hack/lib/metadata.bash: line 4: yq: command not found
```

**Comment by dsimansk** [comment]
@navidshaikh I guess the yq addition needs to go through master to be present?

**Comment by navidshaikh** [comment]
@dsimansk : any changes we do in the PR branch should be the one running in CI, so its not needed at master. Beside, I see that nightly force pushing and causing conflicts, which we'd need to fix.
Beside, can you try adding yq to image and check if its functioning as expected locally ?

**Comment by navidshaikh** [comment]
The setup working fine, parsing from logs
```
========================================
==== KNATIVE INSTALLED SUCCESSFULLY ====
========================================
==== Fri Nov  6 06:23:33 PST 2020
========================================
/go/src/github.com/knative/client
===========================
==== RUNNING E2E TESTS ====
===========================
[..]
```

**Comment by dsimansk** [comment]
Yep, there was one more issue with old path to scripts and hopefully, we can get some decent result.

**Comment by dsimansk** [comment]
Down to one failing test. I wonder why image is resolved by `TEST_IMAGE_TEMPLATE` everywhere else, but not in that expected svc, even in annotation of that particular instance.

```
 === CONT  TestServiceExport
    service_export_test.go:58: export service-revision1 and compare
    service_export_test.go:263: assertion failed: 
        --- expService
        +++ &actSvc
          &v1.Service{
          	TypeMeta:   {Kind: "Service", APIVersion: "serving.knative.dev/v1"},
          	ObjectMeta: {Name: "hello"},
          	Spec: v1.ServiceSpec{
          		ConfigurationSpec: v1.ConfigurationSpec{
          			Template: v1.RevisionTemplateSpec{
          				ObjectMeta: {Name: "hello-rev1", Annotations: {"client.knative.dev/user-image": "registry.build01.ci.openshift.org/ci-op-5rwbc3jk/stable:knative-"...}},
          				Spec: v1.RevisionSpec{
          					PodSpec: v1.PodSpec{
          						Volumes:        nil,
          						InitContainers: nil,
          						Containers: []v1.Container{
          							{
          								Name:    "user-container",
        - 								Image:   "/helloworld:latest",
        + 								Image:   "registry.build01.ci.openshift.org/ci-op-5rwbc3jk/stable:knative-client-test-helloworld", 
```

File: openshift/e2e-common.sh [hunk 1]
```diff
@@ -200,64 +200,15 @@ install_knative_serving_branch() {
   git clone --branch $branch https://github.com/openshift/knative-serving.git /tmp/knative-serving || return 1
   pushd /tmp/knative-serving
 
-  oc new-project $SERVING_NAMESPACE
+  local current_image_format=$IMAGE_FORMAT
```
Comment (rhuss) [inline openshift/e2e-common.sh:203]: Shouldn't be this line after the next one ?

**Comment by dsimansk** [comment]
@rhuss I've seen a comment, that's missing now. :) Anyway to the schroedinger's question then.

No it shouldn't by the following idea.

It stores current value (actual running build) and overrides to nightly variant. That should ensure that serving setup is using the desired nightly value for its images. Finally a bit later set the current value back as we might need it elsewhere. As a result we don't have to override the whole function from the serving setup.
https://github.com/openshift/knative-serving/blob/master/openshift/e2e-common.sh#L91

**Comment by rhuss** [comment]
> No it shouldn't by the following idea.

yeah, sorry. I commented to early before fully reading the code. My bad, therefor I deleted the comment. Sorry for the consufion ...

**Comment by dsimansk** [comment]
Make sure this PR isn't merged by accident.
/hold

**Comment by dsimansk** [comment]
Merged into master.
/close

**Pull request closed without merging**